### PR TITLE
Add optimize command for zend lucene adapter

### PR DESCRIPTION
**Pull request by alexander-schranz**
Add optimize zend lucene command.

### CODE REVIEW

File: Search/OptimizeableInterface.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Massive\Bundle\SearchBundle\Search;
+
+interface OptimizeableInterface
```
Comment (niklasnatter) [inline Search/OptimizeableInterface.php]: think i would call this `OptimizeableAdapterInterface`

File: Command/OptimizeCommand.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    public function configure()
+    {
+        $this->setDescription('Optimize the search index.');
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (!$this->adapter instanceof OptimizeableInterface) {
+            $output->writeln(\sprintf('Adapter "%s" does not support.', \get_class($this->adapter)));
```
Comment (niklasnatter) [inline Command/OptimizeCommand.php]:
```suggestion
            $output->writeln(\sprintf('Adapter "%s" does not support index optimization.', \get_class($this->adapter)));
```

File: Command/OptimizeCommand.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    public function __construct(AdapterInterface $adapter)
+    {
+        parent::__construct(self::$defaultName);
+
+        $this->adapter = $adapter;
+    }
+
+    public function configure()
+    {
+        $this->setDescription('Optimize the search index.');
```
Comment (niklasnatter) [inline Command/OptimizeCommand.php]:
```suggestion
        $this->setDescription('Optimize all search indices.');
```

File: Search/OptimizeableInterface.php (hunk 1 above)
Comment (alexander-schranz) [reply Search/OptimizeableInterface.php]: is okay for me did rename it.

File: Resources/docs/commands.rst [hunk 4]
```diff
@@ -86,3 +86,14 @@ Options:
  - ``index``: Specify index to purge
  - ``all``: Purge all indexes.
  - ``force``: Do not ask for confirmation.
+
+``massive:search:optimize``
+---------------------------
+
+Will optimize the index, this is recommended to be configured as a cronjob when
+using the ``zend_lucene`` adapter, for the other adapters this command does current
+do nothing.
```
Comment (niklasnatter) [inline Resources/docs/commands.rst]:
```suggestion
Optimize all search indices. Affects only indices that are managed with the ``zend_lucene`` adapter at the moment. It is recommended to configure this as cronjob when ``zend_lucene`` adapter.
```

File: Command/OptimizeCommand.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    public function __construct(AdapterInterface $adapter)
+    {
+        parent::__construct(self::$defaultName);
+
+        $this->adapter = $adapter;
+    }
+
+    public function configure()
+    {
+        $this->setDescription('Optimize all search indices. Currently only relevant when using the `zend_lucene` adapter.');
```
Comment (niklasnatter) [inline Command/OptimizeCommand.php]:
```suggestion
        $this->setDescription('Optimize all search indices. Affects only indices that are managed with the zend_lucene adapter at the moment.');
```

**Pull request merged**